src/java/com/unifina/signalpath/blockchain/GetEthereumContractAt.java

```diff
@@ -34,30 +45,84 @@ public void init() {
 		abiParam.setCanConnect(false);
 	}
 
+	protected String getApiUrl(String network, String address){
+		StringBuilder sb = new StringBuilder();
+		sb.append("https://");
+		if(network == null || network.startsWith("main")){
+			sb.append("api");
+		}
+		else{
+			sb.append(network);
+		}
+		sb.append("."+ETHERSCAN_IO+ETHERSCAN_IO_QUERY+address);
+		return sb.toString();
+	}
+	public static String readUtf8StreamToString(InputStream is) throws IOException {
+		ByteArrayOutputStream bos = new ByteArrayOutputStream();
+		byte[] buf = new byte[4096];
+		int len;
+		while((len = is.read(buf)) != -1){
+			bos.write(buf,0,len);
+		}
+		bos.flush();
+		return new String(bos.toByteArray(), StandardCharsets.UTF_8);
+	}
+	protected JsonArray getEthereumAbi(String network, String address) throws IOException {
+		String url = getApiUrl(network, address);
+		URL etherscanapi = new URL(url);
+		URLConnection conn = etherscanapi.openConnection();
+		String json =readUtf8StreamToString(conn.getInputStream());
+		JsonParser parser = new JsonParser();
+		JsonObject response = parser.parse(json).getAsJsonObject();
+		if(response.get("status").getAsString().equals("0")){
+			log.info("No ABI found for address "+address + " on Etherscan for network "+ network);
+			return null;
+		}
+		else{
+			log.info("ABI found for address "+address + " on Etherscan for network "+ network);
+			return parser.parse(response.get("result").getAsString()).getAsJsonArray();
+		}
+	}
+
 	@Override
 	protected void onConfiguration(Map<String, Object> config) {
 		String address = addressParam.getValue();
 		String abiString = MapTraversal.getString(config, "params[1].value");
-
 		ModuleOptions options = ModuleOptions.get(config);
 		ethereumOptions = EthereumModuleOptions.readFrom(options);
+		String network = ethereumOptions.getNetwork();
+		//address != 0x0
+		if(address.length() > 2 && !Numeric.toBigInt(address).equals(BigInteger.ZERO)){
+			if(abiString == null || abiString.trim().equals("") || abiString.trim().equals("[]")){
+				try {
+					//try to pull from etherescan
+					JsonArray etherscan_abi = getEthereumAbi(network, address);
+					abi = new EthereumABI(etherscan_abi);
+					if(etherscan_abi != null){
+						//how do we make the retrieved ABI appear in ABI field on UI?
+						abiParam.value = etherscan_abi.toString();
+					}
```

Comment:
@jtakalai is there some way to display the ABI pulled from Etherscan in abiParam field on canvas?